upgrade to scala 2.12

crispywalrus opened the pull request:
based on my experience with finagle-postgres and travis I've jumped to
2.12.2 to avoid the "build works, travis hangs" issue. I've notably
not upgraded the docs pugings (other than tut) due to the plugin to
autoplugin work. those changes should come in a different PR.

I've also left finagle at 6.43.0 to match finch (and my infrastructure)

crispywalrus committed:
upgrade to scala 2.12

based on my experience with finagle-postgres and travis I've jumped to
2.12.2 to avoid the "build works, travis hangs" issue. I've notably
not upgraded the docs pugings (other than tut) due to the plugin to
autoplugin work. those changes should come in a different PR.

ilya-murzinov commented:
oh, I've already opened #58 with almost the same changes..
but still can't figure out why tut compilation hangs

ilya-murzinov commented:
It works now

crispywalrus committed:
downgrade sbt

other projects get through travis-ci with 2.12.2 but with sbt 0.13.13. so trying varying the sbt version here

crispywalrus commented:
I did see that but opening a PR means I can iterate against the travis-ci builds faster. Using this vs fixing your initial PR ... don't care either way.
